Terraform installation and permissions (#238)
DataDog/datadog-serverless-functions: aws/logs_monitoring/README.md

1. Create a Python3.7 Lambda function using `aws-dd-forwarder-<VERSION>.zip` from the latest [releases](https://github.com/DataDog/datadog-serverless-functions/releases).
1. Save your Datadog API key in AWS Secrets Manager, set environment variable `DD_API_KEY_SECRET_ARN` with the secret ARN on the Lambda function, and add the `secretsmanager:GetSecretValue` permission to the Lambda execution role.
1. If you need to forward logs from S3 buckets, add the `s3:GetObject` permission to the Lambda execution role.
1. If you need to forward custom metrics and traces from your Lambda functions' logs for serverless monitoring (not yet supported in AWS China and GovCloud), attach these [layers](https://github.com/DataDog/datadog-serverless-functions/blob/3639499bf602ea3d04493028aa08d1076cc02234/aws/logs_monitoring/template.yaml#L264) (switch to master branch for the latest layer versions) to the forwarder, and set environment variable `DD_ENHANCED_METRICS` to `false` on the forwarder.
1. Configure [triggers](https://docs.datadoghq.com/integrations/amazon_web_services/?tab=allpermissions#send-aws-service-logs-to-datadog).

</details>

## Terraform Installation

> The forwarder can be installed using Terraform resource [aws_cloudformation_stack](https://www.terraform.io/docs/providers/aws/r/cloudformation_stack.html) as a wrapper on top of the provided CloudFormation template.

Reviewer: ```suggestion
The Forwarder can be installed using Terraform resource [aws_cloudformation_stack](https://www.terraform.io/docs/providers/aws/r/cloudformation_stack.html) as a wrapper on top of the provided CloudFormation template.
```

<details><summary>Sample Configuration</summary>

```tf
variable "dd_api_key" {
  type        = string
  description = "Datadog API key"
}

resource "aws_cloudformation_stack" "datadog-forwarder" {
  name         = "datadog-forwarder"
  capabilities = ["CAPABILITY_IAM", "CAPABILITY_NAMED_IAM", "CAPABILITY_AUTO_EXPAND"]
  parameters   = {
    DdApiKey        = var.dd_api_key
    FunctionName    = "datadog-forwarder"
  }
  template_url = "https://datadog-cloudformation-template.s3.amazonaws.com/aws/forwarder/latest.yaml"
}
```

</details>

## Permissions

> To deploy the CloudFormation stack with the default options, you need to have the permissions below to save Datadog API key as a secret, create a S3 bucket to store the forwarder's zip (source code), and create Lambda functions (including execution roles and log groups).

Reviewer: ```suggestion
To deploy the CloudFormation Stack with the default options, you need to have the permissions below to save your Datadog API key as a secret, create a S3 bucket to store the Forwarder's code (zip file), and create Lambda functions (including execution roles and log groups).
```

<details><summary>IAM Statements</summary>

```json
{
   "Effect": "Allow",
   "Action": [
         "cloudformation:*",
         "secretsmanager:CreateSecret",
         "secretsmanager:TagResource",
         "s3:CreateBucket",
         "s3:GetObject",
         "iam:CreateRole",
         "iam:GetRole",
         "iam:PassRole",
         "iam:PutRolePolicy",
         "iam:AttachRolePolicy",
         "lambda:CreateFunction",
         "lambda:GetFunction",
         "lambda:GetFunctionConfiguration",
         "lambda:GetLayerVersion",
         "lambda:InvokeFunction",
         "lambda:PutFunctionConcurrency",
         "lambda:AddPermission",
         "logs:CreateLogGroup",
         "logs:DescribeLogGroups",
         "logs:PutRetentionPolicy"
   ],
   "Resource": "*"
}
```

</details>

> The CloudFormation stack creates following IAM roles:

Reviewer: ```suggestion
The CloudFormation Stack creates following IAM roles:
```

> - ForwarderRole: The execution role for the Forwarder Lambda function to read logs from S3, fetch Datadog API key from Secrets Manager and write its own logs.

Reviewer: ```suggestion
- ForwarderRole: The execution role for the Forwarder Lambda function to read logs from S3, fetch your Datadog API key from Secrets Manager, and write its own logs.
```
